File: 118 Pascal's Triangle/solution.py
class Solution:
    # @param {integer} numRows
    # @return {integer[][]}
    def generate(self, numRows):
        result = []
        if numRows==0:
            return result
        helper = [1]
        result.append(helper)
        for i in range(numRows-1):
            tmp = helper
            helper = [1]
            for j in range(i):
                helper.append(tmp[j]+tmp[j+1])
            helper.append(1)
            result.append(helper)
        return result

File: 118 Pascal's Triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        if (numRows==0)
            return result;
        vector<int> helper;
        for (int i=0;i<numRows;i++){
            vector<int> tmp = helper;
            helper.push_back(1);
            for (int j=1;j<i;j++)
                helper[j] = (tmp[j-1]+tmp[j]);
            result.push_back(helper);
        }
        return result;
    }
};